src/OSPSuite.SimModelNative/src/FormulaChange.cpp

```diff
@@ -92,6 +92,9 @@ bool FormulaChange::PerformSwitchUpdate(double * y, double time)
 	//---- object to change is not a species - set new formula or value
 	if (_useAsValue)
 	{
+		//add variable indices used by the original quantity formula
+		_quantity->AppendUsedVariables(changedVariablesIndicesUsedInParameters);
+
```

Comment:
first use case: parameter is overwritten by the const value.
Then state variables from the old formula are added to the list